src/Console/ServerBlock.php

```diff
@@ -69,214 +65,141 @@ protected function getHelp()
-			while (($data = fgetcsv($fp, 1000, ',')) !== false) {
-				$domain = $data[0];
-				if (count($data) == 0) {
-					$reason = self::DEFAULT_REASON;
-				} else {
-					$reason = $data[1];
-				}
-				$data = [
-					'domain' => $domain,
-					'reason' => $reason
-				];
-				if (!in_array($data, $newBlockList)) {
-					$newBlockList[] = $data;
-				}
-			}
-
-			foreach ($currBlockList as $blocked) {
-				if (!in_array($blocked, $newBlockList)) {
-					$newBlockList[] = $blocked;
-				}
-			}
+		$newBlockList = $this->blocklist::extractFromCSVFile($filename);
 
-			if ($config->set('system', 'blocklist', $newBlockList)) {
-				$this->out(sprintf("Entries from %s that were not blocked before are now blocked", $filename));
-				return 0;
-			} else {
-				$this->out(sprintf("Couldn't save '%s' as blocked server", $domain));
-				return 1;
-			}
+		if ($this->blocklist->append($newBlockList)) {
+			$this->out(sprintf("Entries from %s that were not blocked before are now blocked", $filename));
+			return 0;
 		} else {
-			throw new Exception(sprintf('The file "%s" could not be opened for importing', $filename));
+			$this->out("Couldn't save the block list");
+			return 1;
 		}
 	}
 
 	/**
-	 * Prints the whole list of blocked domains including the reason
-	 *
-	 * @param IManageConfigValues $config
+	 * Prints the whole list of blocked domain patterns including the reason
 	 */
-	private function printBlockedServers(IManageConfigValues $config)
+	private function printBlockedServers(): void
 	{
 		$table = new Console_Table();
-		$table->setHeaders(['Domain', 'Reason']);
-		$blocklist = $config->get('system', 'blocklist', []);
-		foreach ($blocklist as $domain) {
-			$table->addRow($domain);
+		$table->setHeaders(['Pattern', 'Reason']);
+		foreach ($this->blocklist->get() as $pattern) {
+			$table->addRow($pattern);
 		}
-		$this->out($table->getTable());
 
-		// Success
-		return 0;
+		$this->out($table->getTable());
 	}
 
 	/**
-	 * Adds a server to the blocked list
-	 *
-	 * @param IManageConfigValues $config
+	 * Adds a domain pattern to the block list
 	 *
 	 * @return int The return code (0 = success, 1 = failed)
 	 */
-	private function addBlockedServer(IManageConfigValues $config)
+	private function addBlockedServer(): int
 	{
 		if (count($this->args) < 2 || count($this->args) > 3) {
-			throw new CommandArgsException('Add needs a domain and optional a reason.');
-		}
-
-		$domain = $this->getArgument(1);
-		$reason = (count($this->args) === 3) ? $this->getArgument(2) : self::DEFAULT_REASON;
-
-		$update = false;
-
-		$currBlockList = $config->get('system', 'blocklist', []);
-		$newBlockList = [];
-		foreach ($currBlockList  as $blocked) {
-			if ($blocked['domain'] === $domain) {
-				$update = true;
-				$newBlockList[] = [
-					'domain' => $domain,
-					'reason' => $reason,
-				];
-			} else {
-				$newBlockList[] = $blocked;
-			}
+			throw new CommandArgsException('Add needs a domain pattern and optionally a reason.');
```

Comment:
The reasoning for the block should not be optional on the CLI, as it is not in the admin panel (done by requiring filling out the form field).

Reply:
I just followed the existing check but I guess I can make it explicit, it would save on tests.